dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -622,7 +623,66 @@ public final Response deleteAction(@Context final HttpServletRequest request,
 
         return response;
     } // deleteAction
+    
+    /**
+     * Change the order of the steps in a scheme
+     * @param request                           HttpServletRequest
+     * @param workflowReorderActionStepForm     WorkflowReorderBean
+     * @return Response
+     */
+    @PUT
+    @Path("/reorder/step/{stepId}/order/{order}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response reorderStep(@Context final HttpServletRequest request,
+                                        @PathParam("stepId")   final String stepId, 
+                                        @PathParam("order")    final int order) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
 
+        try {
+            WorkflowStep step = workflowAPI.findStep(stepId);
+            WorkflowScheme scheme = workflowAPI.findScheme(step.getSchemeId());
+            List<WorkflowStep> steps = workflowAPI.findSteps(scheme);
+            IntStream.range(0, steps.size())
+                .filter(i -> steps.get(i).getId().equals(step.getId()))
+                .boxed()
+                .findFirst()
+                .map(i -> steps.remove((int) i));
+
+            int newOrder = (order > steps.size()) ? steps.size():order;
+            steps.add(newOrder, step);
+            
+            int i=0;
+            for(WorkflowStep stepp : steps) {
+                stepp.setMyOrder(i++);
+                workflowAPI.saveStep(stepp);
+            }
```

Comment:
I think we shouldn't have this reorder functionality at a resource leve. I would say at api level.

Reply:
Also let's try to think of resource when possible. Instead of a RPC-style /reorder/ I picture this as a PATCH of the step including only the new order value

Reply:
PATCH /steps/:stepid -d {"order":{newOrderValue}}